src/content/app/entity-viewer/variant-view/variant-image/alternative-allele/AlternativeAllele.tsx

```diff
@@ -0,0 +1,194 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+
+import SequenceLetterBlock from '../sequence-letter-block/SequenceLetterBlock';
+
+import variantGroups from 'src/content/app/genome-browser/constants/variantGroups';
+
+import styles from './AlternativeAllele.module.css';
+
+type Props = {
+  regionSliceStart: number;
+  variantStart: number; // accounts for anchor base in appropriate variant types
+  variantLength: number; // length of reference allele; accounts for anchor base in appropriate variant types
+  alleleType: string;
+  sequence: string;
+  mostSevereConsequence: string;
+  hasAnchorBase: boolean;
+  // onClick: () => void;
+};
+
+const AlternativeAllele = (props: Props) => {
+  const {
+    variantStart,
+    regionSliceStart,
+    alleleType,
+    variantLength,
+    hasAnchorBase,
+    mostSevereConsequence
+  } = props;
+  let { sequence } = props;
+
+  if (hasAnchorBase) {
+    sequence = sequence.slice(1); // exclude the first (anchor) base of the allele
+  }
+
+  const maxReferenceSequenceLength = 41; // FIXME: import this as a constant from another file
+
+  // Maximum alt allele display length is dynamic, and depends on where the variant starts
+  // Alt alleles are allowed to exceed the length of the remaining reference sequence (after variant start)
+  // by five nucleotides
+  const distanceToVariantStart = variantStart - regionSliceStart;
+  const maxAlternativeAlleleLength =
+    maxReferenceSequenceLength - distanceToVariantStart + 5;
+
+  const sequenceLetters = sequence
+    .split('')
+    .slice(0, maxAlternativeAlleleLength);
+  const shouldShowEllipsis = sequence.length > maxAlternativeAlleleLength;
+
+  if (shouldShowEllipsis) {
+    sequenceLetters[sequenceLetters.length - 1] = '…';
+  }
+
+  const colourId = colourToIdMap.get(mostSevereConsequence);
+  const defaultLetterColour = 'var(--color-grey)'; // this is a fallback colour that should never be displayed if everything is working correctly
+  const letterColour = colourId
+    ? colours.get(colourId) ?? defaultLetterColour
+    : defaultLetterColour;
+
+  const sequenceLetterStyle = {
+    ['--sequence-block-color' as string]: letterColour
+  };
+  // if the sequence should end in ellipsis, make the letter block transparent
+  const lastSequenceLetterStyle = shouldShowEllipsis
+    ? {
+        ['--sequence-block-color' as string]: 'transparent',
+        ['--sequence-block-letter-color' as string]: letterColour
```

Comment:
Is it going to be `colour` or `color` in TypeScript?

Reply:
🤷‍♂️ 
I know that we are using `color` in CSS; so I expect the CSS variable `'--sequence-block-letter-color'` to be without the `u`. As for the name of the `letterColour` variable, I would personally love to switch to `color`; but this is not a very strong opinion.